### ISSUE TITLE
Messages from users in private rooms may be received by a catching up client, if created and moved while offline

### ISSUE THREAD

**Issue by Perska** [randomouscrap98/contentapi#134]
Normally, when a message is moved to a private room the client does not have access to, it will remain visible in its original room, as the client does not have access to the new room and is not sent the edit event.

However, if the message is created and rethreaded in the span that the client is offline, once the client reconnects using the last event ID, it will receive the messages that were moved, and gain knowledge of the private room through a message_event.

Timeline of events:
* Client loses connection
* User creates message(s)
* User rethreads the message(s)
* (Unsure: additional events may have to happen before the client reconnects)
* Client reconnects and catches up
* In the catch-up data, the rethreaded messages are present, alongside with the private room object.

TL;DR if connecting with a set last ID, and there is both a message create and rethread in the data that will be sent upon connection, the rethreaded message will be included even it it was moved to a room the client can't access.

**Comment by randomouscrap98** [comment]
Perska, I updated oboy. Would you be willing to check over there to see if your specific situation is fixed? It's ok if not, I'm going to write a complex unit test to test this scenario

**Comment by randomouscrap98** [comment]
I added a test specifically for this scenario which fails with the old method but works in the new method. If you still find issues, please let me know, thank you very much for submitting an issue
